resolver/serviceResolver.go

```diff
@@ -269,6 +269,82 @@ func (resolver *serviceResolver) SignMessage(userIp string, request requests.Sig
 
 }
 
+// SetSecurityModeNoExpire gets the user's guardian, verifies the codes and then sets the SecurityMode
+func (resolver *serviceResolver) SetSecurityModeNoExpire(userIp string, request requests.SetSecurityModeNoExpireMessage) (*requests.OTPCodeVerifyData, error) {
+	userAddress, err := sdkData.NewAddressFromBech32String(request.UserAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	addressBytes := userAddress.AddressBytes()
+	resolver.userCritSection.Lock(string(addressBytes))
+	defer resolver.userCritSection.Unlock(string(addressBytes))
+
+	userInfo, err := resolver.getUserInfo(addressBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	ctxGetGuardianData, cancelGetGuardianData := context.WithTimeout(context.Background(), resolver.requestTime)
+	defer cancelGetGuardianData()
+	guardianData, err := resolver.httpClientWrapper.GetGuardianData(ctxGetGuardianData, request.UserAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	guardianAddrBytes, err := resolver.pubKeyConverter.Decode(guardianData.ActiveGuardian.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	verifyCodeData, err := resolver.checkAllowanceAndVerifyCode(userInfo, request.UserAddr, userIp, request.Code, request.SecondCode, guardianAddrBytes)
+	if err != nil {
+		return verifyCodeData, err
+	}
+
+	err = resolver.secureOtpHandler.SetSecurityModeNoExpire(request.UserAddr)
+
+	return nil, err
```

Comment:
can be simplified to:
```suggestion
	return nil, resolver.secureOtpHandler.SetSecurityModeNoExpire(request.UserAddr)
```

Reply:
fixed